index.html

```diff
@@ -0,0 +1,394 @@
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let name = (document.getElementById("name").value).toString();
+  let task = (document.getElementById("task").value).toString();
+  let date = (document.getElementById("date").value).toString();
+  let obj = { id:arr1.length+1, Name:name, Task:task, Date:date};
+  arr1.push(obj);
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+function del(id) {
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  for (let i =0 ; i< arr1.length;i++)
+  {
+    if(arr1[i].id == id)
+    {
+        arr1.splice(i,1);
+        document.getElementById("tb").deleteRow(i+1);
+        i--;
+    }
+  }
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+
+function edit() {
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let t = document.getElementById("editID").value
+  for (let i =0 ; i< arr1.length;i++)
+  {
+    if(arr1[i].id == t)
+    {
+        let name = document.getElementById("name").value;
+        let task = document.getElementById("task").value;
+        let date = document.getElementById("date").value;
+
+        arr1[i].Name = name;
+        arr1[i].Task = task;
+        arr1[i].Date = date;
+    }
+  }
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+function getInf(self)
+{
+    let data = JSON.parse(self.dataset.test);
+    console.log("data",data.Date.split("/").join("-"));
+    document.getElementById("name").value = data.Name;
+    document.getElementById("task").value = data.Task;
+    document.getElementById("date").value = data.Date.split("/").join("-");
+    document.getElementById("editID").value = data.id;
+}
+function searchTask()
+{
+  let word = document.getElementById("search").value;
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let arr2 = [];
+  for(let i =0; i<arr1.length;i++)
+  {
+    console.log(arr1[i].Task.includes(word));
+    if(arr1[i].Task.includes(word))
+    {
+      arr2.push(arr1[i]);
+    }
+  }
+  console.log(arr2);
+  document.getElementById("tb").innerHTML = "";
+  let tr = [];
+  tr.push('<tr><td>STT</td>');
+  tr.push('<td>Tên</td>');
+  tr.push('<td>Công việc</td>');
+  tr.push('<td>Ngày làm</td>');
+  tr.push('<td>Sửa</td>');
+  tr.push('<td>Xóa</td></tr>');
+  for (let i =0 ; i< arr2.length;i++) {
+    let option = document.createElement("option");
+    tr.push('<tr><td id="index">'+arr2[i].id+'</td>');
+    tr.push('<td id="nameintd">'+arr2[i].Name+'</td>');
+    tr.push('<td id="taskintd">'+arr2[i].Task+'</td>');
+    tr.push('<td id="dateintd">'+arr2[i].Date+'</td>');
+    tr.push(`<td><button data-test='${JSON.stringify(arr2[i])}' onclick='getInf(this)'>Sửa</button></td>`);
+    tr.push('<td><button onclick="del('+ arr2[i].id +')">Xóa</button></td></tr>');
+  }
+  let tb = document.getElementById("tb");
+  tb.innerHTML = tr.join("");
+}
+var temp = false;
+function sortByCate(flag)
+{
+
+  if(flag==0){
```

Comment:
remove all if else statement jus call  sortTableByTD(flag);
but should have a enum for save flag value. for instance: 
const SORT_FLAG = {
name: 0,
date:1, ...}

use to compare: (flag === SORT_FLAG.name)